service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RemovalToolController.java

```diff
@@ -134,14 +143,19 @@ public void handleSubmittedEvent(@RequestBody CallbackRequest callbackRequest) {
         CaseData caseDataBefore = getCaseDataBefore(callbackRequest);
 
         Optional<StandardDirectionOrder> removedSDO = orderService.getRemovedSDO(
-            caseData.getHiddenStandardDirectionOrders(), caseDataBefore.getHiddenStandardDirectionOrders()
+            caseData.getRemovalToolData().getHiddenStandardDirectionOrders(),
+            caseDataBefore.getRemovalToolData().getHiddenStandardDirectionOrders()
         );
         Optional<HearingOrder> removedCMO = orderService.getRemovedCMO(
-            caseData.getHiddenCMOs(), caseDataBefore.getHiddenCMOs()
+            caseData.getRemovalToolData().getHiddenCMOs(), caseDataBefore.getRemovalToolData().getHiddenCMOs()
         );
         Optional<AdditionalApplicationsBundle> removedApplication = applicationService.getRemovedApplications(caseData
-                .getHiddenApplicationsBundle(),
-            caseDataBefore.getHiddenApplicationsBundle());
+            .getRemovalToolData().getHiddenApplicationsBundle(),
+            caseDataBefore.getRemovalToolData().getHiddenApplicationsBundle());
+
+        Optional<DocumentReference> removedApplicationForm =
+            (caseData.getC110A().getSubmittedForm() != caseDataBefore.getC110A().getSubmittedForm())
```

Comment:
Just want to check, will we always have a C110a application at this stage on the case data?

Reply:
.getC110A() will always be a thing (as it's a JsonUnwrapped container class), and as it's the submitted event the checks in about-to-submit would make sure that there was a submittedForm beforehand, so generally this condition will always be true, but is there just incase nothing has happened (either no removal or no document being there in the place, as in both of these they'll be equal!)